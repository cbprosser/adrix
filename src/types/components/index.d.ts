export * from './AdvancedSearchForm';
export * from './MTGStaticTypes';
export * from './MultipleCardDisplay';
export * from './SimpleCardSearchBar';
export * from './Symbol';
