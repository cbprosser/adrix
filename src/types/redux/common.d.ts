export type RequestStatus =
  | {
      message?: string;
      requestId?: string;
      status: 'idle' | 'calling';
      success?: boolean;
    }
  | {
      message: string;
      requestId?: string;
      status: 'idle' | 'calling';
      success: false;
    };

export type APISlice<T extends string = string> = {
  requests: Record<T, RequestStatus>;
};
