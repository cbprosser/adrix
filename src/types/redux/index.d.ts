export * from './slices';

export * from './Root';
