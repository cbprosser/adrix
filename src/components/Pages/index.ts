export * from './AdvancedSearch';
export * from './Home';
export * from './Results';
