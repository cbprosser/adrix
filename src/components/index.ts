export * from './AppHeader';
export * from './CardDisplay';
export * from './Pages';
export * from './SimpleCardSearchBar';
