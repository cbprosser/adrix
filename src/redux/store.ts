import { configureStore } from '@reduxjs/toolkit';
import { cardsSlice } from './slices/cards';

export const store = configureStore({
  reducer: {
    cards: cardsSlice.reducer,
  },
});
