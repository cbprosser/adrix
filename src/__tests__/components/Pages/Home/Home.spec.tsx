import { render } from '@testing-library/react';
import { Home } from '../../../../components';
import { initialCardsState, useAppSelector } from '../../../../redux';

jest.mock('../../../../components/SimpleCardSearchBar', () => ({
  SimpleCardSearchBar: (props: any) => (
    <div {...props}>SimpleCardSearchBar</div>
  ),
}));

const mockDispatch = jest.fn();

jest.mock('../../../../redux/hooks', () => ({
  useAppSelector: jest.fn(),
  useAppDispatch: jest.fn(() => mockDispatch),
}));

const mockSelector = useAppSelector as jest.Mock;

describe('Home Suite', () => {
  beforeEach(() => {
    mockSelector.mockReturnValue(initialCardsState);
  });

  it('Should match the snapshot', () => {
    const { asFragment } = render(<Home />);

    expect(asFragment()).toMatchSnapshot();
  });
});
