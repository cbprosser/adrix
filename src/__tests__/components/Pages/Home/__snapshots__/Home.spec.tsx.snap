// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Suite Should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1ss2ei0-MuiContainer-root"
    style="display: flex; flex-direction: column; justify-content: center; align-items: center; height: 100vh;"
  >
    <div>
      SimpleCardSearchBar
    </div>
  </div>
</DocumentFragment>
`;
